Standards/scs-0217-v1-baseline-cluster-security.md

```diff
@@ -0,0 +1,148 @@
+different APIs, authorization and authentication procedures or even Pod privilege mechanisms.
+In order to mitigate these problems, different steps and mechanisms could be used to increase
+the security of a Kubernetes setup.
+
+## Design Considerations
+
+### External CA
+
+Kubernetes provides an API to provision TLS certificates that can be signed by a CA.
+This CA can be controlled by the cluster provider, which enables much more tight control
+over the clusters communication and therefore also better controllable security.
+
+In order to do this, the CA certificate bundle needs to be added to the trusted certificates
+of the server.
+To provide a certificate, the following steps need to be undertaken:
+
+1. Create a CSR
+2. Send the CSR manifest to the k8s API
+3. Approve the CSR
+4. Sign CSR with your CA
+5. Upload the signed certificate to the server
+
+This certificate could now be used by a user in a pod in order to provide a trusted certificate.
+
+It is also possible for the Kubernetes controller manager to provide the signing functionality.
+To enable this, `--cluster-signing-cert-file` and `--cluster-signing-key-file` need to be set with
+a reference to the CA keypair, which was used in the previous example to sign a CSR.
+
+### Protected Kubernetes endpoints
+
+In order to secure a Kubernetes cluster, the protection of endpoints is important.
+To do this, different approaches can be taken.
+
+#### TLS for all internal/API traffic
+
+It is already expected by Kubernetes that all API communication internally is encrypted with TLS.
+Nevertheless, some endpoints of internal components could be/will be exposed without the necessary
+encryption, which could lead to weak points in the system.
+A list of the default service endpoints can be seen in the following table
+
+| Protocol | Port Range  | Purpose                 | Notes                                                                                   |
+|----------|-------------|-------------------------|-----------------------------------------------------------------------------------------|
+| TCP      | 6443*       | Kubernetes API Server   | -                                                                                       |
+| TCP      | 2379-2380   | etcd server client API  | -                                                                                       |
+| TCP      | 10250       | Kubelet API             | -                                                                                       |
+| TCP      | 10251/10259 | kube-scheduler          | 10251 could be insecure before 1.13, after that only the secure port 10259 is available |
+| TCP      | 10252/10257 | kube-controller-manager | 10252 could be insecure before 1.13, after that only the secure port 10257 is available |
+| TCP      | 10255       | Read-Only Kubelet API   | HTTP port for GET requests                                                              |
```

Comment:
Mhh I don't really know. As far as my understanding goes, this port is used to request metrics from the kubelet.
So if the port would be disabled, there could be problems when retrieving metric data. On the other hand, it should be possible to use the normal secured endpoint to retrieve these metrics.

Personally, I would declare it as insecure and advise against using it. Since it is normally set to `0`, I think most people won't even set it.

Reply:
I proposed a remark, that would work for me. Retaining the table as is would also work for me btw.